### ISSUE TITLE
[polymer-editor-service] It would be nice if ../ always meant "look at the bower directory"

### ISSUE THREAD

**Issue by rictic** [Polymer/tools#452]
> Migrated from https://github.com/Polymer/atom-plugin/issues/3
> On September 29th, 2016, @notwaldorf said:

Context: if you have an enormous folder with a billion elements in it because you work on Polymer Elements and that's how you cope, auto-complete looks at the sibling directories rather than the bower deps, which may be woefully out of date.

*Originally posted at Sun, 16 Oct 2016 08:13:26 GMT at https://github.com/Polymer/polymer-editor-service/issues/2*

**Comment by rictic** [comment]
> On October 4th, 2016, @TimvdLippe said:

I just downloaded and followed the instructions in the README and for me the linter was not recognizing these `../` paths. Since `bower install` creates `bowercomponents/`, in a usage project the imports are expected to be `bower_components/paper-spinner/paper-spinner.html` instead of `../paper-spinner/paper-spinner.html`. Now this is an example project, so in general I think it is fine. Just my 2 cents of my initial impression of relative url imports.


*Originally posted at Sun, 16 Oct 2016 08:14:29 GMT at https://github.com/Polymer/polymer-editor-service/issues/2#issuecomment-254033750*

**Comment by rictic** [comment]
Yeah, this would make the editors easier to use and less fragile, as you wouldn't have to make your workspace root exactly right.

We could keep the current behavior for the analyzer by adding a new URL resolver that searches for a bower_components directory when resolving a ../ URL anywhere.

What do you think @justinfagnani ?


*Originally posted at Wed, 19 Oct 2016 14:05:04 GMT at https://github.com/Polymer/polymer-editor-service/issues/2#issuecomment-254822594*

**Comment by justinfagnani** [comment]
@rictic I need to understand what's going on here. The PackageUrlResolver should already be doing this.


*Originally posted at Wed, 19 Oct 2016 20:19:59 GMT at https://github.com/Polymer/polymer-editor-service/issues/2#issuecomment-254928722*

**Comment by rictic** [comment]
`PackageUrlResolver` only does that rewriting when the path is reaching out of the basedir. This is a proposal for doing it opportunistically, whenever it looks like it would help.


*Originally posted at Thu, 20 Oct 2016 00:19:41 GMT at https://github.com/Polymer/polymer-editor-service/issues/2#issuecomment-254977099*
